pyo3-build-config/src/impl_.rs

```diff
@@ -32,7 +32,9 @@ pub fn cargo_env_var(var: &str) -> Option<String> {
 /// Gets an external environment variable, and registers the build script to rerun if
 /// the variable changes.
 pub fn env_var(var: &str) -> Option<OsString> {
-    println!("cargo:rerun-if-env-changed={}", var);
+    if cfg!(feature = "resolve-config") {
+        println!("cargo:rerun-if-env-changed={}", var);
+    }
```

Comment:
@davidhewitt let me know if this is problematic or if there's a better way to suppress this output in maturin.

Reply:
This seems like a sensible way to do it to me!